streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/SuppressImpl.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streams.kstream.Suppress;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+import java.time.Duration;
+
+public final class SuppressImpl<K, V> implements Suppress<K, V> {
+    private IntermediateSuppressionImpl<K, V> intermediateSuppression = null;
+    private TimeDefinition<K, V> timeDefinition = (context, k, v) -> context.timestamp();
```

Comment:
nit: just define two static `TimeDefinition<K, V>` and `TimeDefinition<Windowed<K>, V>`, one with context.timestamp and one with window end time?